packages/mobile/src/web3/contracts.ts

```diff
@@ -4,30 +4,49 @@
  * Leaving the name for recognizability to current devs
  */
 import { ContractKit, newKitFromWeb3 } from '@celo/contractkit'
-import { RpcWallet } from '@celo/contractkit/lib/wallets/rpc-wallet'
 import { sleep } from '@celo/utils/src/async'
 import { call, delay, select } from 'redux-saga/effects'
 import { ContractKitEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
 import { ErrorMessages } from 'src/app/ErrorMessages'
 import { DEFAULT_FORNO_URL } from 'src/config'
-import { isProviderConnectionError } from 'src/geth/geth'
+import { getGethInstance, isProviderConnectionError } from 'src/geth/geth'
 import { waitForGethInitialized } from 'src/geth/saga'
 import { navigateToError } from 'src/navigator/NavigationService'
 import Logger from 'src/utils/Logger'
 import { getHttpProvider, getIpcProvider } from 'src/web3/providers'
 import { fornoSelector } from 'src/web3/selectors'
 import Web3 from 'web3'
-import { IpcProvider } from 'web3-core'
+import { RNGethWallet } from '../geth/RNGethWallet'
 
 const TAG = 'web3/contracts'
 const KIT_INIT_RETRY_DELAY = 2000
 const CONTRACT_KIT_RETRIES = 3
 
-let ipcProvider: IpcProvider | undefined
-let gethWallet: RpcWallet | undefined
+let gethWallet: RNGethWallet | undefined
 let contractKit: ContractKit | undefined
 
+function* initWallet() {
+  ValoraAnalytics.track(ContractKitEvents.init_contractkit_get_wallet_start)
+  const wallet = new RNGethWallet(yield call(getGethInstance))
```

Comment:
The wallet does not need a gethInstance in DataSaver mode, right?

Reply:
Resolving based on our chat